### ISSUE TITLE
Newest pre release resets town claims upon restart

### ISSUE THREAD

**Issue by ILikeOrangeJuice** [TownyAdvanced/Towny#7337]
### What steps will reproduce the problem?

• We use 0.100.1.0 on our live server, copied our files to our test server (both run 1.20.1 paper), tested all out, all worked fine.
• We then proceeded to turn off the test server and apply the newest version 0.100.1.24, which started with no errors or issues.
• We modified the new settings and permissions to our need, saved all, restarted the server, no error came up again.
• We tried the new cede command which behaved weird, because no matter how far away it was, it only gave the message of being too close to a townblock, we then checked our /t here settings and noticed that all claims but the 3-4 we claimed to test it were gone. We then checked other cities and they still hat their outpost position, homeblock position, you could tp, they had all their bonus chunks, but everything was unclaimed eg. town size 0/150 [Bought:140/10000] (10 claims default)
• We tried to restart another time, but again, no error happened while loading but this time we catched some upon using commands which are the following:
https://pastebin.com/LMsmQr98
https://pastebin.com/8A5HrB5U
https://pastebin.com/LEtsk7Fw

Restoring our backup on the server fixed the issues and the claims are fine.

### What is the expected output?

That it works as it used to be and that our claims do not reset upon restarting.

### Towny version

0.100.1.24

### Server version

paper 1.20.1 #193

### Please use Pastebin.com to link the following files

1. Your full server startup from the logs\latest.log:
First startup with the new version:
https://mclo.gs/nJX7SxO
Startup after adjusting all needs:
https://mclo.gs/01my2eC
2. Your Towny config.yml:
https://mclo.gs/rNEscEk
4. Your townyperms.yml:
https://mclo.gs/hAeT9Jl
6. Your log's error:
https://pastebin.com/LMsmQr98
https://pastebin.com/8A5HrB5U
https://pastebin.com/LEtsk7Fw

(optional) If this is to do with permissions,
7. Your permissions file:
We mostly use the listed above townyperms file but those are the towny permissions we added extra with Luckperms to a group each user has access to:
towny.command.town.claim.*
towny.command.town.invite.*
towny.cheat.bypass
towny.command.resident.tax
towny.command.town.claim.fill
towny.command.town.claim.town.multiple
towny.command.town.join
towny.command.town.leave
towny.command.town.list
towny.command.town.new
towny.command.town.set.mapcolor
towny.command.town.set.name
towny.command.towny.map
towny.town.spawn.public
towny.town.spawn.town

But we don't think it is an permission issue as it always has worked and the added perms have nothing to do with our issue we believe.

(optional) If this is to do with chat,
8. Your chatconfig.yml:
9. Your channels.yml:

**Comment by LlmDl** [comment]
Does your test server's world have an identical UUID to the live server?

**Comment by ILikeOrangeJuice** [comment]
Yes, it is 1:1 the same map as it does get mirrored, I just, to make 150% sure, did the same steps on the live server rn as none was active, with the same result unfortunately, but rolling back fixed it again. So it shouldn't be an issue with the map.

**Comment by LlmDl** [comment]
No one else has had any issue updating Towny throughout the pre-releases. That's why I figured your world data might not be exact. Did you change your world name in any way on the test server?

When you say it is mirrored, have you made sure the UUID for the world is a match?

**Comment by ILikeOrangeJuice** [comment]
Yes it is a straight 1:1 copy of the entire server, database etc. so it is identical. I have unfortunately no other idea too. I could retry it this night again with the .2 release and see if that somehow magically fixed it and give feedback again?

**Comment by ILikeOrangeJuice** [comment]
Ok actually I had a min and non was on so I tested the newest release .2 on the live server, same issue, it loads fine, but once all is loaded up the claims are gone, you can also see them disappearing on the dynmap, I went on and checked towns, they had no claimed plots again and well, when I did /t spawn errors occured again. Here is the latest log: https://mclo.gs/LTnhez4

**Comment by LlmDl** [comment]
What is the world name that your towns are in?

**Comment by ILikeOrangeJuice** [comment]
Another detail I noticed, when I did test the new cede command I claimed a long way of claims on my server, even tho the backup is restored, my town still has those claims and they work without issues with the old version, so it the claim infact did work and we could also rule out an issue with the db if something changed there? As it worked until I restarted, so it must be something in the newer version which breaks the loading of claimed plots on the startup, at least this is how it seems to me.

**Comment by LlmDl** [comment]
Show me what your TOWNY_WORLDS table looks like.

**Comment by ILikeOrangeJuice** [comment]
I exported it to yml to post it easier:
https://pastebin.com/D3bvfrhv

I noticed there are old worlds in it tho, currently we have the following worlds:
Farmwelt - no towny world, our farmworld.
Farmwelt-Nether - no towny world, our farmworld.
Farmwelt-Ende - no towny world, our farmworld.
spawn - no towny world, old random world.
SpawnWelt - No towny world, spawn world.
Claimwelt - towny claimable world.

plots - doesn't exist anymore
world - doesn't exist anymore
kingdom - doesn't exist anymore
Towny - doesn't exist anymore

Should I just delete their entries? However I doubt that is the cause, as it would have caused issues beforehand, as those are very old worlds and gone since a long time haha.

**Comment by LlmDl** [comment]
Old worlds dont really matter. You can remove any that aren't present on the server anymore.

Could I see an output of your TOWNY_TOWNS table in the same YAML format?

**Comment by ILikeOrangeJuice** [comment]
Of course! I also just to take an eventual step off, and included the townytownblocks one, however that one is several mb big so I had to include them as files in a zip. 

Here is a virustotal scan in case of worriness :D https://www.virustotal.com/gui/file/6a69cadad457bb5d7b268ee8ff0a2cbf4dbc320075eb86444b789d3fcc741ace?nocache=1

[townyTowns+Townblocks.zip](https://github.com/TownyAdvanced/Towny/files/14844658/townyTowns%2BTownblocks.zip)

**Comment by LlmDl** [comment]
I'd really rather not have to get a whole zip file.

Could I see an output of your TOWNY_TOWNS table in the same YAML format?

**Comment by ILikeOrangeJuice** [comment]
https://hastebin.com/share/robuxelihi.yaml

**Comment by LlmDl** [comment]
In your towny config.yml can you set `debug_mode: true` and then show me your complete startup log.

**Comment by ILikeOrangeJuice** [comment]
This is the startup from .1 (the working version) enabled with the debug_mode: https://mclo.gs/1SCYeGg
This is the startup (+ stopping) from .2 (the broken version) enabled with the debug_mode: https://mclo.gs/Z6sAf2e

**Comment by LlmDl** [comment]
Could you both test using this jar: https://cdn.discordapp.com/attachments/669247087329673228/1226528928890490981/towny-0.100.2.1.jar?ex=66251914&is=6612a414&hm=3d2e06280b473933a2f9bc218d310c2ddd13eb50d8a80e8c7448c28f4fd8b520&

**Comment by ILikeOrangeJuice** [comment]
Tested it with the provided jar and it seems to work so far on a quick test.
Installed the new version and used /t after to check and the number of claimed chunks was correct and not 0 like before.
I restarted a second time after adjusting the townyperms and config to our liking with all the changes that came between .1 and .2 and it worked after as well. I then claimed new chunks, restartet again and they were still correctly claimed and counted in eg. /t.

If anything deeper needs to be tested, let me know, but as of now it seems to work!:) 

Thanks for the fix! Also thanks @Parthodys for the help identifying the patch it occured!
